src/complex.rs

```diff
@@ -132,6 +133,38 @@ impl<T: Clone + Float> Complex<T> {
         let (r, theta) = self.to_polar();
         Complex::from_polar(&(r.sqrt()), &(theta/two))
     }
+    
+    /// Raises a number to a floating point power.
+    #[inline]
+    pub fn powf(&self, value: T) -> Complex<T> {
+        let (r, theta) = self.to_polar();
+        Complex::from_polar(&(r.powf(value)), &(theta*value))
+    }
+    
+    /// Returns the logarithm of the number with respect to an arbitrary base.
+    #[inline]
+    pub fn log(&self, value: T) -> Complex<T> {
+        let (r, theta) = self.to_polar();
+        // Comment for code review:
+        // There is for sure a faster way to get the constant, but I don't know how to do that in this generic context 
+        // Any hints?
+        let e = T::one().exp(); 
+        Complex::new(r.log(value), theta * e.log(value))
+    }
+    
+    /// Raises a floating point number to a complex power.
+    #[inline]
+    pub fn expb(&self, value: T) -> Complex<T> {
+        // Comment for code review:
+        // Same as value.powf(self) - if it would exist - but I would prefer to have this version
+        // of the function around since I feel like I would look for it in the complex
+        // type. But we would also be able to provide it by creating a trait for real numbers.
+        // Your thoughts?
```

Comment:
Silly me, full `complex^complex` should probably be just `pow`.
But I still lean toward `expf` here.